### PR TITLE
Remove push trigger from promotion pipeline

### PR DESCRIPTION
**Pull request by kakkoyun**
Simplifies the social media promotion pipeline by removing the push-triggered promotion mode. Posts are now promoted only via the nightly cron (6 AM UTC) or manual dispatch, eliminating the need for deduplication logic between push and schedule triggers.

## Summary

The promotion workflow previously had three triggers: push (for newly added posts), nightly cron (for scheduled posts), and manual dispatch. This change removes the push trigger entirely, as `claude-code-action` does not support the `push` event type. Posts merged on their publish day after the 6 AM cron will need to be promoted manually via the workflow dispatch button.

## Key Changes

- **Removed push mode** from `scripts/find-promotable-posts.sh`: deleted the `push` case that filtered for newly added files via `git diff --diff-filter=A`
- **Removed deduplication logic**: deleted `file_added_date()` function and the check that skipped posts added today (which was meant to avoid double-promotion between push and cron triggers)
- **Updated schedule mode**: now scans all posts without the dedup check
- **Simplified workflow trigger configuration** (`.github/workflows/promote-post.yml`):
  - Removed `push` event trigger
  - Made `post_path` input optional for manual dispatch (blank = promote everything due today)
  - Updated conditional logic to handle manual dispatch with/without a specific post path
  - Removed `fetch-depth: 0` (no longer needed for git log dedup)
- **Updated documentation** (CLAUDE.md): clarified that there are now only two triggers (cron + manual), explained the limitation, and removed dedup explanation
- **Updated test suite** (`scripts/test-find-promotable-posts.sh`):
  - Removed all push mode test cases (`_push_today`, `_push_future`, `_push_draft`, etc.)
  - Removed dedup-specific test (`_sched_dedup`)
  - Updated schedule mode test to expect promotion of today-added posts (no longer skipped)
  - Migrated edge-case tests from push to schedule mode
- **Updated helper script** (`scripts/posts-publishing-today.sh`): removed mention of dedup logic in comments

## Implementation Details

The schedule mode now unconditionally promotes any post with `publishDate == today` that passes validation (not draft, `promote: false` honored). The workflow dispatch button can target a specific post (skipping date validation) or run in schedule mode to promote everything due today.

https://claude.ai/code/session_01SQQJ7r1gKC2SY2W1RQ9VDZ

### CODE REVIEW

File: .github/workflows/promote-post.yml [hunk 1]
```diff
@@ -34,19 +36,18 @@ jobs:
...
-          if [ "${{ github.event_name }}" = "workflow_dispatch" ]; then
+          if [ "${{ github.event_name }}" = "workflow_dispatch" ] && [ -n "${{ github.event.inputs.post_path }}" ]; then
+            # Manual run targeting one specific post (skips the date check).
             POSTS=$(scripts/find-promotable-posts.sh manual "${{ github.event.inputs.post_path }}")
-          elif [ "${{ github.event_name }}" = "schedule" ]; then
-            POSTS=$(scripts/find-promotable-posts.sh schedule)
           else
-            POSTS=$(scripts/find-promotable-posts.sh push)
+            # Daily cron, or a manual run with no post_path: promote everything due today.
+            POSTS=$(scripts/find-promotable-posts.sh schedule)
```
[the remark this answers is not in the document]
Comment (kakkoyun) [reply .github/workflows/promote-post.yml]: Good catch — fixed in 38f2d79. `post_path` now reaches the Find step via an `env:` var (`POST_PATH`) and the script references `"$POST_PATH"`, so bash treats it as opaque data instead of expanding `$(...)`/backticks. `github.event_name` is a GitHub-controlled enum and the `dry_run` input is only used in `if:` expression contexts (not a shell), so neither is an injection vector.

---
_Generated by [Claude Code](https://claude.ai/code)_
